src/formBuilder/formBuilder.jsx

```diff
@@ -57,6 +57,9 @@ const FormBuilder = props => {
   const [formFields, setFields] = useState(null);
   const [formErrors, setFormErrors] = useState([]);
 
+  const [customSecondaryBtn, setCustomSecondaryBtn] = useState(secondaryBtn);
+  const [customLifecycleFns, setCustomLifecycleFns] = useState([]);
+
```

Comment:
Need some feedback on this. Added state vars in order to set custom Tx lifecycle functions based on form state changes (triggered by fields). For example, this allows to properly handle a WalletConnect session otherwise, it will remain open after a proposal is submitted/cancelled, causing unexpected behaviours on future connections

Reply:
This works for me.